modules/kafka/kafka-dest-worker.c

```diff
@@ -61,30 +61,59 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
   int block_flag = _is_poller_thread(self) ? 0 : RD_KAFKA_MSG_F_BLOCK;
-
-  if (rd_kafka_produce(owner->topic,
+  rd_kafka_topic_t *topic;
+  GString *topic_name_new = g_string_sized_new(128);
```

Comment:
this is a good candidate for scratch buffers which would avoid a per-message allocation.